src/common/DirectoryTreeView.tsx

```diff
@@ -0,0 +1,79 @@
+import TreeView from '@mui/lab/TreeView'
+import TreeItem from '@mui/lab/TreeItem'
+import { useState, useCallback, useEffect } from 'react'
+import { DirectoryArrayMetadata } from 'server/routes/v1/deployment'
+import ChevronRightIcon from '@mui/icons-material/ChevronRight'
+import ExpandLessIcon from '@mui/icons-material/ExpandLess'
+import useNotification from './Snackbar'
+import { getEndpoint } from '../../data/api'
+
+export default function DirectoryTreeView({
```

Comment:
I think we need to be careful about which components end up in `/common`. This is a component designed for a specific purpose. It is not a generic, reusable component and therefore shouldn't be in this directory imo.